plugins/woocommerce/src/Admin/API/Patterns.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\API;
+
+use Automattic\WooCommerce\Blocks\BlockPatterns;
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\Patterns\PTKClient;
+use Automattic\WooCommerce\Blocks\Patterns\PTKPatternsStore;
+use Automattic\WooCommerce\StoreApi\Exceptions\RouteException;
+use Exception;
+use WP_Error;
+use WP_HTTP_Response;
+use WP_REST_Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product controller
+ *
+ * @internal
+ */
+class Patterns {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/patterns',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_pattern' ),
+					'permission_callback' => function () {
+						return is_user_logged_in();
+					},
+				),
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'update_patterns' ),
+					'permission_callback' => function () {
+						return is_user_logged_in();
+					},
+				),
+			)
+		);
+	}
+
+	/**
+	 * Fetch a single pattern from the PTK to ensure the API is available.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response
+	 * @throws RouteException If the patterns cannot be fetched.
+	 */
+	public function get_pattern() {
+		$ptk_client = Package::container()->get( PTKClient::class );
```

Comment:
Do you think it would be better to inject this dependency through the constructor?

Reply:
Not sure we can in this case, since the endpoints are initialized like [this](https://github.com/woocommerce/woocommerce/blob/31a3ee9f233b1fec194fadbceec4953be27ce5f0/plugins/woocommerce/src/Admin/API/Init.php#L96-L171).